src/days/day03.rs

```diff
@@ -42,18 +39,17 @@ impl Map {
     }
 }
 
-type Position = (u32, u32);
-type Slope = Position;
-
-fn apply_slope(slope: Slope, mut position: Position) -> Position {
-    position.0 += slope.0;
-    position.1 += slope.1;
+struct Position(u32, u32);
 
-    position
+impl Position {
+    fn apply_slope(&mut self, slope: &Position) {
+        self.0 += slope.0;
+        self.1 += slope.1;
+    }
```

Comment:
Before, we were both mutating `position` and returning it, which was unnecessary.  This just modifies it in-place, which works well enough here.  We could make it immutable, but there isn't really any point.